packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetServiceWithoutSearch.java

```diff
@@ -187,6 +206,10 @@ public Optional<T> updateAsset(final T asset, final Schema.Permission hasWritePe
 			throw new IllegalArgumentException("Cannot update a non-temporary asset to be temporary");
 		}
 
+		if (projectService.isProjectPublic(projectId)) {
+			asset.setPublicAsset(true);
```

Comment:
How does this get set to false if the project is no longer public?

Reply:
Doh